### PR TITLE
Image handling

### PR DESCRIPTION
**Pull request by mauilion**
This is a couple of commits to improve image handing.
The first decouples building the builder-image from identifying whether it needs built.
It also implements a way to push the builder-image tagged with the commit that most recently changed it to the velero org.
The logic is: If the Dockerfile has local changes than build a new image Else try to pull a build-image if that fails build one anyway. 

The second fixes the `make clean` target so that we use the builder-image (if present) to clean the mod cache before cleaning the dirs.

### CODE REVIEW

**Comment by nrb** [comment]
@mauilion I've been holding off on reviewing since it says WIP - let us know when you're ready for a look!

File: .github/workflows/push-builder.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Master CI
```
[the remark this answers is not in the document]
Comment (mauilion) [reply .github/workflows/push-builder.yml]: your right! it's fixed.

File: .github/workflows/push-builder.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: build-image
+
+on:
+  push:
+    branches: [ main ]
```
Comment (ashish-amarnath) [inline .github/workflows/push-builder.yml:5]: Depending on branch `main` requires #2677 to be merged before this PR.

**Comment by ashish-amarnath** [comment]
Considering #2677 is not fully ready for merge, should re update the github actions to run on the `master` branch instead and make the branch change as part of #2677 

@carlisia and @nrb WDYT?

File: .github/workflows/push-builder.yml (hunk 2 above)
Comment (nrb) [reply .github/workflows/push-builder.yml:5]: I am ok with this depending on `master` and rebasing 2677 on this PR if you would like to change this back, @mauilion

**Comment by mauilion** [comment]
changed and rebased! Thanks!

**Comment by nrb** [comment]
@carlisia and @ashish-amarnath looks ok to me, back to you!

**Pull request merged**